RatFinder/app/src/main/java/team45/ratfinder/controller/FirebaseObjectConverter.java

```diff
@@ -9,16 +12,36 @@
  */
 
 public class FirebaseObjectConverter {
-
+public int test;
     static RatSighting getRatSighting(Map<String, Object> map, String uniqueKey) {
+        Log.d("test", map.get("Incident Zip").toString()+ "here");
+
+        int incidentZip = map.get("Incident Zip").toString().equals("") ? 0: ((Long) (map.get("Incident Zip"))).intValue();
+        double latitude = map.get("Latitude").toString().equals("") ? 0: ((double) (map.get("Latitude")));
+        double longitude = map.get("Longitude").toString().equals("") ? 0: ((double) (map.get("Longitude")));
         return new RatSighting(uniqueKey,
-                (String) map.get("Created Date"),
+                (long) map.get("Created Date"),
                 (String) map.get("Location Type"),
-                (int) ((long) map.get("Incident Zip")),
+                incidentZip,
                 (String) map.get("Incident Address"),
-                (String)map.get("City"),
+                (String) map.get("City"),
                 (String) map.get("Borough"),
-                (double) map.get("Latitude"),
-                (double) map.get("Longitude"));
+                latitude,
+                longitude);
+    }
+    static Map<String, Object> sightingToFireBase(RatSighting sighting) {
+        Map<String, Object> data = new HashMap<String, Object>();
+        data.put(("Created Date"), sighting.getCreatedDate());
+        data.put(("Location Type"), sighting.getLocationType());
+        data.put(("Incident Zip"), sighting.getIncidentZip());
+        data.put(("Incident Address"), sighting.getIncidentAddress());
+        data.put(("City"), sighting.getCity());
+        data.put(("Borough"), sighting.getBorough());
+        data.put(("Latitude"), sighting.getLatitude());
```

Comment:
both latitude and longitude need to be cast as doubls when inserting into the database so that firebase doesn't assume they are longs

Reply:
yeah that needs to be fixed

Reply:
nvm the map is <string, object>, firebase will see that sighting.getLongitude is already a double, there is no need to cast. if you cast, you will see the android studio will tell you that the cast is redundant.